File: examples/objective-c-demo/objective-c-demo/PlayerViewController.m
#import "PlayerViewController.h"

@interface PlayerViewController (){
    AVPlayer *_avplayer;
    AVPlayerViewController *_avplayerController;
    MLYAVPlayerPlugin *_plugin;
}
@end

@implementation PlayerViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [self setupUI];
    
    _plugin = [[MLYAVPlayerPlugin alloc] init];
    NSError *error = nil;
    
    [MLYDriver initializeAndReturnError:  &error :^(MLYDriverOptions * option) {
        [[option client] setId: @"cegh8d9j11u91ba1u600"];
        
        [_plugin adapt:_avplayerController];
    }];
    
    if(error != nil){
        NSLog(@"MLYDriver initialize err:%@",error.description);
    }
    NSURL *url = [NSURL URLWithString:@"https://vsp-stream.s3.ap-northeast-1.amazonaws.com/HLS/raw/SpaceX.m3u8"];
    AVPlayerItem *item = [AVPlayerItem playerItemWithURL:url];
    [_avplayer replaceCurrentItemWithPlayerItem:item];
    [_avplayer play];
}

- (void)setupUI {
    _avplayer = [[AVPlayer alloc] init];
    _avplayerController = [AVPlayerViewController new];
    _avplayerController.showsPlaybackControls = true;
    _avplayerController.view.backgroundColor = UIColor.darkGrayColor;
    _avplayerController.player = _avplayer;
    
    [self addChildViewController:_avplayerController];
    [[self view] addSubview:_avplayerController.view];
}
  
@end

File: examples/swift-demo/swift-demo/PlayerViewController.swift
import AVFoundation
import AVKit
import MLYSDK
import SnapKit
import UIKit

class PlayerViewController: UIViewController {
    lazy var player = AVPlayer()
    lazy var playerViewController = {
        let controller = AVPlayerViewController()
        controller.showsPlaybackControls = true
        controller.view.translatesAutoresizingMaskIntoConstraints = false
        controller.view.backgroundColor = .darkGray
        controller.player = self.player
        return controller
    }()

    lazy var plugin = MLYAVPlayerPlugin()

    override func viewDidLoad() {
        super.viewDidLoad()
        self.setupUI()
        do {
            try MLYDriver.initialize { options in
                options.client.id = DemoConfig.defaultConfig.id
                options.server.host.fqdn = DemoConfig.defaultConfig.server
                options.debug = true
            }
            self.plugin.adapt(self.playerViewController)
        } catch {
            print(error)
        }
        let url = URL(string: DemoConfig.defaultConfig.url)!
        let playerItem = AVPlayerItem(url: url)
        self.player.replaceCurrentItem(with: playerItem)
        self.player.play()
    }

    func setupUI() {
        self.view.backgroundColor = .black
        self.addChild(self.playerViewController)
        self.view.addSubview(self.playerViewController.view)
        self.playerViewController.view.snp.makeConstraints { make in
            make.edges.equalToSuperview()
        }
    }
}
